### PR TITLE
Setup loading sound process to use fully qualified file paths

### LINKED ISSUE #128: 🚧Setup loading sound process to use fully qualified file paths
### I have done the items below . . .

- [X] I have updated the title by replacing the '**_<title_**>' section.

### Description

Currently the sound loader only has the ability to load sounds in the content directory.  Change the sound loading process to also take in fully qualified file paths to anywhere on disk to load a sound.

### Acceptance Criteria

**This issue is finished when:**
- [x] Throws exception if the file does not exist for fully qualified file paths
- [x] Throws exception if the resolved file path to a sound content item does not exist
- [x] Is able to take in a non-path content name with and without and extension and the extension is ignored
- [x] Scene added to testing application to test sounds
- [x] Code documentation added if required
- [x] Unit tests added
- [x] All unit tests pass

### ToDo Items

- [x] Draft pull request created and linked to this issue
- [X] Priority label added to issue (**_low priority_**, **_medium priority_**, or **_high priority_**)
- [X] Issue linked to the proper project
- [X] Issue linked to proper milestone
- [x] Unit tests have been written and/or adjusted for code additions or changes

### Issue Dependencies

_No response_

### Related Work

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #176: 🧪QA Testing Velaptor
### To Do List:

- [X] I have linked all issues that are getting tested for this release
- [X] QA testing label added to issue (🧪qa testing)

### Version Of Velaptor Being Tested

v1.0.0-preview.7

### Relevant Information

This release requires testing of 1 new feature and also added testing functionality for existing sound behavior.

### **Added features:**
1. Rendering of rectangles
    - This new feature adds the ability for a user to easily render rectangles to the screen with various attributes.
    These attributes are:
        - Full and empty rectangles
        - Border thickness of empty rectangles
        - Width and height
        - Position
        - Corner radius for each corner independently
        - If the type of gradient is set to none, the color of the full rectangle and the border color of the empty rectangle
        - Horizontal color gradients for full and empty rectangles
        - Vertical color gradients for full and empty rectangles
        - Starting color of either gradient type
        - Ending color of either gradient type

### Issues To Test

- [x] #153
    - Just make sure that all other testing application functionality works as normal 
- [x] #75 
    - This will involve using the new **rectangle testing scene** that has been added to the testing application
- [x] #131 
    - This will involve using the new **sound testing scene** that has been added to the testing application

### QA Testing Dependencies

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>

<details><summary>📄Description📄</summary>

This pull request is for preview release **_v1.0.0-preview.7_**
</details>

<h2 style="font-weight:bold" align="center">✅Development Checklist✅</h2>

<details open><summary>🌳Branching🌳</summary>

Does the name of the head(source) branch for this pull request have the correct **_preview feature branch_** syntax?

Syntax: _preview/feature/&lt;issue-id&gt;-&lt;description&gt;_  
Example: _preview/feature/123-my-preview-change_
- [x] Yes
- [ ] No
 
Does the name of the base(target) branch for this pull request have the correct **_preview branch_** syntax?

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No

Is the head(source) branch for this pull request created from a branch with the correct **_preview branch_** syntax?

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No
</details>

<details open><summary>🐛Bugs🐛</summary>

Contains Bug Fix(es)
- [x] Yes
    - [x] A bug-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>💣Breaking Change(s)💣</summary>

Contains [breaking change(s)](https://docs.microsoft.com/en-us/dotnet/core/compatibility/#modifications-to-the-public-contract)
- [ ] Yes
- [x] No
</details>

<details open><summary>✨Enhancements✨</summary>

Contains enhancement(s) that add a feature or behavior.
- [x] Yes
    - [x] An enhancement-label label has been added to the pull request.
- [ ]  No
</details>

<details open><summary>⚙️Workflow (CI/CD) Changes⚙️</summary>

Contains changes to workflow files. These changes can only done by the project maintainers.
- [ ] Yes
    - [ ] A workflow-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>📃Documentation Updates📃</summary>

Contains changes that require documentation updates to code docs or **Velaptor** documentation.
- [x] Yes
    - [x] I have updated the documentation accordingly.
    - [x] A documentation-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>🧪Unit Testing🧪</summary>

My change requires unit tests to be written.
- [x] Yes
- [ ] No
</details>

<details open><summary>🧪Manual Testing🧪</summary>

I have manually tested my changes. (This can be done by using project named **_VelaptorTesting_**).
- [x] Yes
- [ ] No
</details>

<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] Pull request title matches the title of the linked issue.
- [x] The **_[add version here]_** text in the pull request description was replaced with the version.
- [x] Associated issue exists and is linked to this pull request.
    - One issue per pull request.
- [x] My code follows the coding style of this project.
    - This is enforced by the *.editorconfig* files in the project and displayed as warnings.  If there is an edge case with coding style that should be ignored or changed, reach out and let's discuss it.
- [x] All tests passed locally.
    - Status checks are put in place to run unit tests every single time a change is pushed to a pull request.  This does not mean that the tests pass in both the local and CI environment.
- [x] A preview-label label has been added to the pull request.

### CODE REVIEW

**Comment by CalvinWilkinson** [comment]
Found a small bug where trying to stop and unloaded sound causes and exception.

File: Testing/VelaptorTests/Content/SoundLoaderTests.cs [hunk 1]
```diff
@@ -19,34 +21,46 @@ namespace VelaptorTests.Content
...
-        private readonly string oggSoundFilepath;
+        private const uint OggSoundId = 1234u;
+        private const uint Mp3SoundId = 5678u;
+        private readonly string oggSoundFilePath;
+        private readonly string mp3SoundFilePath;
         private readonly Mock<IPathResolver> mockSoundPathResolver;
         private readonly Mock<ISoundFactory> soundFactory;
+        private readonly Mock<IFile> mockFile;
         private readonly Mock<IPath> mockPath;
         private readonly Mock<IReactable<DisposeSoundData>> mockDisposeSoundReactable;
```
Comment (kselena) [inline Testing/VelaptorTests/Content/SoundLoaderTests.cs:36]: Should this have two >>?

File: Velaptor/Content/Sound.cs [hunk 2]
```diff
@@ -153,7 +153,15 @@ public float PlaySpeed
         /// Stops the sound.
         /// </summary>
         /// <remarks>This will set the time position back to the beginning.</remarks>
-        public void Stop() => this.sound.Stop();
+        public void Stop()
+        {
+            if (this.sound.Unloaded)
```
Comment (CalvinWilkinson) [inline Velaptor/Content/Sound.cs:158]: NOTE: This was a bug fix here.
This was stopping the sound when it was already unloaded which was causing an exception within the CASL library.

**Review by CalvinWilkinson — COMMENTED** [review]
PR Approved

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by kselena — APPROVED** [review]
